front/tests/chat/e2e_eval.ts

```diff
@@ -0,0 +1,194 @@
+/**
+ * Run an eval of our chat app with a test set of questions See:
+ * https://www.notion.so/dust-tt/Design-Doc-Chat-Eval-0b849d4c66564f1e9c0c31d537c96f78
+ *
+ * Use `--verbose` to print the successes as well as the failures Usage: from
+ * front directory `./admin/eval.sh [--verbose] [--n NUMBER_OF_QUESTIONS]`
+ *
+ * By default, runs on all questions. Use `--n NUMBER_OF_QUESTIONS` to change
+ * that (e.g. smaller evals). This is to allow fast testing of the eval code
+ *
+ * As the eval code may need to change over time and has been written with the
+ * intention of being reused for other kinds of chat evals, the ability to not
+ * run it on all questions is useful, especially since the questions test set is
+ * likely to become sizeable
+ *
+ * The chat sessions created by the eval are not saved in the database, to avoid
+ * flooding the database with test data that can be considered noisy.
+ *
+ */
+
+import { Storage } from "@google-cloud/storage";
+
+import {
+  cloneBaseConfig,
+  DustProdActionRegistry,
+} from "@app/lib/actions/registry";
+import { runAction } from "@app/lib/actions/server";
+import { newChat } from "@app/lib/api/chat";
+import { Authenticator } from "@app/lib/auth";
+import { Ok } from "@app/lib/result";
+import { RunType } from "@app/types/run";
+
+type ChatEvalInputType = {
+  question: string;
+  rules: string[];
+  answer: string;
+  time?: string;
+};
+
+type EvalResultType = {
+  rule: string;
+  rule_respected: "YES" | "NO" | "UNCLEAR";
+  explanation?: string;
+};
+
+async function chatEval(
+  numberOfQuestions: number | undefined,
+  verbose = false
+) {
+  /* Auth */
+  const workspaceId = process.env.LOCALHOST_WORKSPACE_ID as string;
```

Comment:
Surprising name? I didn't see it being referenced in the commands I've seen flying around?

Why do we need a local worskpace? couldn't it be run entirely against prod?

Reply:
Sorry for not explaining this var

As the tool is run on our machines via cli (main use case when iterating I think), since I call  Authenticator.internalBuilderForWorkspace(workspaceId) => needs the workspaceId of my local workspace

I should indeed clarify that this requires devs to have their localhost workspaceid in their .env.local

is there a simpler way using our "dev" prod workspace? at the time I'm unclear how it would work when run locally

Reply:
Would you be up for writing a small runbook covering in full details how to use the script?

Given my current understanding it is my belief that any local workspace would work as they all get redirected to prod dust when in local dev setup right?

Reply:
Yes, any workspace you have in your local postgres
The runbook is [here](https://www.notion.so/dust-tt/d151a5226dba4f8fbe094365da5f00d4?v=69d6cd00975e4bc196dc9877c7c1cb39&p=8b720ae0d2cf4b6c8fe61592b90d7289&pm=s), i'll update it with the info about this var